cdbot/cogs/maths.py

```diff
@@ -143,31 +142,55 @@ async def challenge(self, ctx: Context, number: int = 1):
         return await ctx.send(embed=embed)
 
     @command()
-    async def latex(self, ctx: Context, expression: str):
-        """Render a LaTeX expression."""
+    async def latex(self, ctx: Context, *, expression: str):
+        """
+        Render a LaTeX expression with https://quicklatex.com/
+        """
+        await self.latexRender(ctx, expression)
+
+    async def latexRender(self, ctx: Context, expression: str):
         channel = ctx.channel.id if type(ctx) is Context else ctx.id
 
         if channel in constants.BLOCKED_CHANNELS:
             return await ctx.send(
                 "\N{NO ENTRY SIGN} You cannot use this command in this channel!", delete_after=10
             )
 
-        options = {
-            "auth": {"user": "guest", "password": "guest"},
-            "latex": expression,
-            "resolution": 900,
-            "color": "969696",
-        }
+        # Code and regexes taken from https://quicklatex.com/js/quicklatex.js
+        # aiohttp seems to URL-encode things in a way quicklatex doesn't like
+
+        if expression.startswith("...latex") or expression.startswith(":latex"):
+            return
+
+        formula = expression.replace("%","%25").replace("&","%26")
+
+        preamble = constants.LATEX_PREAMBLE.replace("%","%25").replace("&","%26")
+        
+        body =        'formula=' + formula
+        body = body + '$$$$&fsize=50px'
+        body = body + '&fcolor=969696'
+        body = body + '&mode=0';
+        body = body + '&out=1';
+        body = body + '&errors=1';
+        body = body + '&preamble=' + preamble;
```

Comment:
A couple things worth noting here:

Firstly, `urllib.parse.urlencode` is a handy method for urlencoding parameters. You can use it like this:
```python
from urllib.parse import urlencode

...

body = urlencode({
    'formula': formula + '$$$$',
    'fcolor': 969696,
    'mode': 0,
    'out': 1,
    'errors': 1,
    'preamble': constants.LATEX_PREAMBLE
})
```
Secondly, semicolons aren't required in Python so you can omit those.

Reply:
The issue with this is for some strange reason quicklatex.com doesn't like URL-encoded parameters...

Reply:
semicolons fixed :) ~was copying from the JS implementation, to work out how they got round the URL-encoding issue...~